quantum/keycode_string.c

```diff
@@ -0,0 +1,404 @@
+            return;
+
+        case QK_PERSISTENT_DEF_LAYER ... QK_PERSISTENT_DEF_LAYER_MAX:  // PDF(layer) key.
+            append_unary_keycode(PSTR("PDF"), number_string(QK_PERSISTENT_DEF_LAYER_GET_LAYER(keycode), 10));
+            return;
+
+        // Mod-tap MT(mod,kc) key. This implementation formats the MT keys where
+        // one modifier is applied. For MT keys with multiple modifiers, the mod
+        // arg is written numerically as a hex code.
+        case QK_MOD_TAP ... QK_MOD_TAP_MAX: {
+            uint8_t mods = QK_MOD_TAP_GET_MODS(keycode);
+            const bool is_rhs = mods > 15;
+            mods &= 15;
+            if (mods != 0 && (mods & (mods - 1)) == 0) {  // One mod is set.
+                append_char(is_rhs ? 'R' : 'L');
+                append_P(mod_names[biton(mods)]);
+                append_P(PSTR("_T("));
+            } else {
+                append_P(PSTR("MT("));
+                append_number(mods, 16);
+                append_char(',');
+            }
+            append_keycode(QK_MOD_TAP_GET_TAP_KEYCODE(keycode));
+            append_char(')');
+        }   return;
+
+        case QK_TAP_DANCE ... QK_TAP_DANCE_MAX:  // Tap dance TD(i) key.
+            append_unary_keycode(PSTR("TD"), number_string(QK_TAP_DANCE_GET_INDEX(keycode), 10));
+            return;
+
+#ifdef UNICODE_ENABLE
+        case QK_UNICODE ... QK_UNICODE_MAX:  // Unicode UC(codepoint) key.
+            append_unary_keycode(PSTR("UC"), number_string(QK_UNICODE_GET_CODE_POINT(keycode), 16));
+            return;
+#elif defined(UNICODEMAP_ENABLE)
+        case QK_UNICODEMAP ... QK_UNICODEMAP_MAX:  // Unicode Map UM(i) key.
+            append_unary_keycode(PSTR("UM"), number_string(QK_UNICODEMAP_GET_INDEX(keycode), 10));
+            return;
+
+        case QK_UNICODEMAP_PAIR ... QK_UNICODEMAP_PAIR_MAX: {  // UP(i,j) key.
+            const uint8_t i = QK_UNICODEMAP_PAIR_GET_UNSHIFTED_INDEX(keycode);
+            const uint8_t j = QK_UNICODEMAP_PAIR_GET_SHIFTED_INDEX(keycode);
+            append_P(PSTR("UP("));
+            append_number(i, 10);
+            append_char(',');
+            append_number(j, 10);
+            append_char(')');
+        }   return dest;
+#endif
+#ifdef MOUSEKEY_ENABLE
+        case MS_BTN1 ... MS_BTN8:  // Mouse button keycode.
+            append_P(PSTR("MS_BTN"));
+            append_number(keycode - (MS_BTN1 - 1), 10);
+            return;
+#endif  // MOUSEKEY_ENABLE
+
+        case KB_KEYCODE_RANGE:  // Keyboard range keycode.
+            append_P(PSTR("QK_KB_"));
+            append_number(keycode - QK_KB_0, 10);
+            return;
+
+        case USER_KEYCODE_RANGE:  // User range keycode.
+            append_P(PSTR("QK_USER_"));
+            append_number(keycode - QK_USER_0, 10);
+            return;
+    }
+    // clang-format on
+
+    append_number(keycode, 16); // Fallback: write keycode as hex value.
+}
+
+const char* keycode_string(uint16_t keycode) {
```

Comment:
I would kind of prefer having "get" at the start of the function.  This wold align more with some of the other string macros that are present (`get_u8_str`, etc).  And makes the function a bit more obvious. 

```suggestion
const char* get_keycode_string(uint16_t keycode) {
```

Reply:
Good point about consistency. Done!